### ISSUE TITLE
Release 3.7.0

### ISSUE THREAD

**Issue by Jaifroid** [kiwix/kiwix-js#943]
Although we had a release relatively recently (for this Repo), I feel that full-text search is a biggie and worth a relatively early release (post festivities).

- [ ] Create and complete a few minor PRs that would make this release easier: #913, #914, #942, possibly #915 
- [ ] Create PR to update the changelog file
- [ ] Rename existing milestones (increment their names), create a new 3.8 milestone and move open issues to this new 3.8
- [ ] Final tests
- [ ] Add release date to changelog and squash/merge PR
- [ ] Create a tag on git
- [ ] Generate unsigned extensions (should be automatically triggered on GitHub Actions by the tag creation)
- [ ] Check that the PWA version is correctly set in moz-extension.kiwix.org (should be automatically triggered by tag creation)
- [ ] Upload the unsigned extensions to Stores:
  - [ ] Firefox - https://addons.mozilla.org/fr/developers/addon/kiwix-offline/edit
    - [ ] Check that the minimum Firefox version is correct
  - [ ] Chrome
  - [ ] Edge - https://partner.microsoft.com/en-us/dashboard/microsoftedge/
  - [ ] Ubuntu touch - https://open-store.io
- [ ] Prepare the source code for version 3.7 (changes should be made to 6 files, including create_all_packages)
- [ ] When available, copy the signed extension binaries to https://download.kiwix.org/release/browsers/
  + [ ] Firefox - [xpi downloadable from extension page](https://addons.mozilla.org/fr/firefox/addon/kiwix-offline/)
  + [ ] Chrome 
  + [ ] Edge - [check version is published](https://microsoftedge.microsoft.com/addons/detail/kiwix/jlepddlenlljlnnhjinfaciabanbnjbp), then use [direct link for Edge-specific crx](https://edge.microsoft.com/extensionwebstorebase/v1/crx?response=redirect&prod=chromiumcrx&prodchannel=&x=id%3Djlepddlenlljlnnhjinfaciabanbnjbp%26installsource%3Dondemand%26uc)
- [ ] Check that the implementation on kiwix.github.io/kiwix-js/ is synchronized automatically with master
- [ ] Publicize these new versions

**Comment by Jaifroid** [comment]
@kelson42 After I've finished the final tests and merged the CHANGELOG PR (see checkboxes above), are you happy for me to publish this release?

**Comment by Jaifroid** [comment]
Manual tests completed. Everything appears to be working as planned.

**Comment by Jaifroid** [comment]
@kelson42 The unsigned version of the Chrome extension is [here](https://download.kiwix.org/nightly/2023-01-04/kiwix-chrome-unsigned-extension-3.7.0.zip). Could you kindly upload to the Chrome store, for which I don't have the credentials? I'll do the Edge, Firefox and Ubuntu Touch stores.

**Comment by Jaifroid** [comment]
@kelson42 Yes, this is due to the new libzim binary. What happened is that the dev version, which is unminified, got included in the final release, which I wasn't aware of. The Firefox submission complained about this, so I deleted the dev versions and resubmitted, and it was accepted.

I'll make an issue to add code to remove the dev versions from packaged apps for future versions. Is this sufficient, or would you prefer to have the unsigned extension without the dev binaries, and resubmit package before the Chrome review is approved? I have a version from which I've deleted those files now, and can upload if necessary. On the other hand, a 5MB download for the update is not enormous for most users, compared to typical app sizes nowadays.

Let me know what you think.

**Comment by Jaifroid** [comment]
Issue created: https://github.com/kiwix/kiwix-js/issues/953.

**Comment by Jaifroid** [comment]
Having merged the PR fixing #953, I re-ran the build script from GitHub Actions, which has automatically updated the release files for firefox-os and ubuntu-click (on https://download.kiwix.org/release/), so they are now a more acceptable size, though still a bit larger than before due to the new (minified) libzim binaries.

**Comment by Jaifroid** [comment]
Just waiting for the Chrome and Edge stores to update now before publicizing this version.

**Comment by kelson42** [comment]
@Jaifroid I think we are all good?!

**Comment by Jaifroid** [comment]
Although the Edge Store entry is updated, it still seems to be serving version 3.6..., or at least mine is not updating despite refreshes, etc. Not sure how long the underlying database takes to update...

**Comment by Jaifroid** [comment]
I spoke too soon, it's just updated to 3.7. So we're good to go. I'll do some publicity on reddit and mastodon (a bit later today).

**Comment by Jaifroid** [comment]
Done!
